File: backend/financial/catalog/api/src/main/java/br/com/systec/opusfinancial/api/service/BankService.java
package br.com.systec.opusfinancial.api.service;


import br.com.systec.opusfinancial.api.filter.FilterBank;
import br.com.systec.opusfinancial.api.domain.Bank;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public interface BankService {

    Bank findById(UUID bankId);

    Page<Bank> findByFilter(FilterBank bankFilter);

    Bank findByCode(String code);

    Map<UUID, Bank> findByIds(Iterable<UUID> ids);

    List<Bank> findAll();
}

File: frontend/src/financial/bank/service/bank.service.ts
import { PageResponse } from '@/common/model/page.response';
import api from '@/service/config/api';
import { Bank } from '../model/bank.model';

class BankService {
    _endpoint: string = '/api/v1/banks';

    public async findByFilter(page: number = 0, keyword: string = ''): Promise<PageResponse<Bank>> {
        const response = await api.get(`${this._endpoint}/filter?page=${page}&keyword=${keyword}`);
        return response.data;
    }
}

export default new BankService();
